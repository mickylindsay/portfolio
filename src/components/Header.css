.Header {
  margin: 0px;
  padding: 0px;
  font-size: 1em;
  font-weight: 400;
  line-height: 1.5;
}

h1 {
  margin: 0px;
}

h2 {
    display: inline-block;
    margin: 0px;
    font-size: .75em;
}

h3 {
    margin: 5px;
    font-size: .5em;
}

.header-body {
  padding: 1rem 1.5rem;
  color: #FFF;
}

#title {
  font-weight: 600;
  line-height: 1.125;
}

.slider {
  height: calc(15px + 2vmin);
  color: #FFF;
  width: 100%;
}

.slider h2 {
  position: absolute;
}

.slider-active {
  width: 70%
}

@media only screen and (max-width: 675px){
  .slider-active {
    width: 100%
  }
}

.slider-inactive {
  width: 30%;
}

@media only screen and (max-width: 675px){
  .slider-inactive {
    width: 0%;
  }
}

#slider-left {
  background: #F90;
  float: left;
}

#slider-right{
  background: #333;
  float:right;
}

#slider-right h2 {
  float: right;
  right: 20px;
}

#slider-left h2 {
  overflow: hidden;
  left: 20px;
}

.slider-active h2 {
  opacity: 1;
}

.slider-inactive h2 {
  opacity: 0;
}
