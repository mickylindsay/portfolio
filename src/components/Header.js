import React from 'react';
import './Header.css';

const Header = props => {
  const sliderClass = (phase) => { return `slider ${phase ? 'slider-active' : 'slider-inactive'}` }

  return (
    <div className="Header">
      <div className="header-body">
          <h1 id="title">{props.phase ? "Micky Lindsay" : "demiSardonic"}</h1>
          <h3 id="title">One part portfolio website, two parts personal website</h3>
      </div>
      <div className="slider" onClick={props.flip}>
        <div id="slider-left" className={sliderClass(props.phase)}><h2>Professonal Programmer and Amateur Public Speaker</h2></div>
        <div id="slider-right" className={sliderClass(!props.phase)}><h2>Amateur Falconeer and Professonal Liar</h2></div>
      </div>
    </div>
  )
}

export default Header
