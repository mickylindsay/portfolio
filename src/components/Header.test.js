import React from 'react';
import { shallow } from 'enzyme';
import Header from './Header';

describe('Header component', () => {
  test('test click slider', () => {
    const mockFlip = jest.fn();

    const header = shallow(<Header flip={mockFlip}/>);
    header.find('.slider').first().simulate('click');
    expect(mockFlip.mock.calls.length).toEqual(1);
  });

  it('matches snapshot with phase true', () => {
    const wrapper = shallow(<Header phase={true}/>);
    expect(wrapper).toMatchSnapshot();
  });

  it('matches snapshot with phase false', () => {
    const wrapper = shallow(<Header phase={false}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
