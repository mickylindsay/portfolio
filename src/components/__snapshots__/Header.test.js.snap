// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component matches snapshot with phase false 1`] = `
<div
  className="Header"
>
  <div
    className="header-body"
  >
    <h1
      id="title"
    >
      demiSardonic
    </h1>
    <h3
      id="title"
    >
      One part portfolio website, two parts personal website
    </h3>
  </div>
  <div
    className="slider"
  >
    <div
      className="slider slider-inactive"
      id="slider-left"
    >
      <h2>
        Professonal Programmer and Amateur Public Speaker
      </h2>
    </div>
    <div
      className="slider slider-active"
      id="slider-right"
    >
      <h2>
        Amateur Falconeer and Professonal Liar
      </h2>
    </div>
  </div>
</div>
`;

exports[`Header component matches snapshot with phase true 1`] = `
<div
  className="Header"
>
  <div
    className="header-body"
  >
    <h1
      id="title"
    >
      Micky Lindsay
    </h1>
    <h3
      id="title"
    >
      One part portfolio website, two parts personal website
    </h3>
  </div>
  <div
    className="slider"
  >
    <div
      className="slider slider-active"
      id="slider-left"
    >
      <h2>
        Professonal Programmer and Amateur Public Speaker
      </h2>
    </div>
    <div
      className="slider slider-inactive"
      id="slider-right"
    >
      <h2>
        Amateur Falconeer and Professonal Liar
      </h2>
    </div>
  </div>
</div>
`;
